pkg/util/runtime/features_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runtime
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFeatures(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Empty query string", func(t *testing.T) {
+		err := ParseFeatures("")
+		assert.NoError(t, err)
+		assert.False(t, FeatureEnabled(FeatureExample))
+		assert.Empty(t, EncodeFeatures())
```

Comment:
This is in the same package as the code so you could check if `featureGates` is nil. Is this a proxy for that check?

Reply:
This was also just to make sure we got the same thing out as we put in, and that it didn't fall over if there was no actual features.